corehq/apps/locations/models.py

```diff
@@ -310,15 +310,21 @@ def get_from_user_input(self, domain, user_input):
 
     def filter_path_by_user_input(self, domain, user_input):
         """
-        Returns a queryset including all locations matching the user input
-        and their children. This means "Middlesex" will match:
-            Massachusetts/Middlesex
-            Massachusetts/Middlesex/Boston
-            Massachusetts/Middlesex/Cambridge
-        It matches by name or site-code
-        """
-        direct_matches = self._user_input_filter(domain, user_input)
-        return self.get_queryset_descendants(direct_matches, include_self=True)
+        Returns a queryset based on user input, filtering on the full path
+          - "middlesex" will return child locations in that county
+          - matching happens by name or site-code
+          - users can filter across multiple nodes by splitting with a slash
+          - a query segment can be wrapped in quotes, to require an exact name match
+        Refer to TestFilterPath for example usages.
+        """
+        query = self
+        for part in filter(None, user_input.split('/')):
+            if part.startswith('"') and part.endswith('"'):
```

Comment:
what happens when user is entering the name and has type the partial name with the starting quotes only?
for ex: "middle

we would search for `middle` as the default behaviour until the user puts the closing quotes?

Reply:
@mkangia, yes exactly. `filter_by_user_input()` calls `_user_input_filter()`, so it preserves current behaviour until the final `"` is typed.

Reply:
I think it used to do exact search even if you just have the first quote.

Reply:
That's correct.  I prefer this behavior for two reasons - one, it's what google does, so arguably more familiar.  More importantly, this is a typeahead filter, so if the user has input `"Mass`, it's likely they just haven't finished the word yet, and it makes more sense to show `Massachusetts` than no results, unless they've put the closing quote, indicating that they _do_ want an exact match on `"Mass"`

Reply:
(you can see this in test cases here: https://github.com/dimagi/commcare-hq/pull/26671/files#diff-4a69b47c01ede698484ccc328abbc5c4R192-R194)

Reply:
Sounds good to me, just wanted to ensure that my understanding is correct that behaviour has now changed.
I just feel that when a user has decided to do an exact match and is typing the location name for exact search, anything the HQ shows in that time period is not useful for the user i.e if i am searching for Massachusetts, a person like me would type `"ma` and then will type slowly trying to remember what the spelling is, in that interim whatever result HQ is showing me, doing all the effort to find matches and descendants is not useful for me because the reason i am doing an exact search is because the default search isn't working for my search. It's just unnecessary search and display that the user is ignoring.
This probably will be reduced since you have changed to just locations by name and not their descendants included.

Reply:
> will type slowly trying to remember what the spelling is

This sounds similar to what Ethan's saying - if you don't remember the spelling, and we're showing similar results, you might notice your option in the list and then not need to spell it out. And if that's not the case and you do ignore those interim results, that's fine.